### PR TITLE
External safe nav

### PR DESCRIPTION
**Pull request by SalBakraa**
Correctly parse a separated `safe_nav` operator through the external Scanner. This branch has merged all the previous pull requests into it.

### CODE REVIEW

**Comment by fwcd** [comment]
Thanks for the PRs, could you look into fixing the failing test?

**Comment by SalBakraa** [comment]
The thing is I don't know how to fix it. It occurs due to the anonymous function getting parsed as a function call with a trailing lambda. I think that is due to that fact that you can't exclude the `fun` keyword from call expressions, so maybe if https://github.com/tree-sitter/tree-sitter/pull/246 gets passed we can parse the statement accurately.

Should I remove this test for the mean time? or is there another way you'd like the for the test to be dealt with?

**Comment by fwcd** [comment]
If this doesn't regress anything that parses on the current `main` branch, I would be fine with removing the test case and merging it as is (and also creating an issue so we don't lose track of this bug). The [web editor](https://fwcd.dev/tree-sitter-kotlin/) seems to parse the test case just fine however, therefore I would ideally like to get this fixed before merging.

File: test/corpus/functions.txt [hunk 1]
```diff
@@ -171,3 +171,106 @@ fun test() {
                 (simple_identifier))
               (value_argument
                 (simple_identifier)))))))))
+
+==================
+Anonymus function
```
Comment (fwcd) [inline test/corpus/functions.txt:176]:
```suggestion
Anonymous function
```

File: test/corpus/functions.txt [hunk 2]
```diff
@@ -171,3 +171,106 @@ fun test() {
...
+
+(source_file
+  (anonymous_function)
+  (property_declaration
+    (variable_declaration
+      (simple_identifier))
+    (anonymous_function)))
+
+==================
+Anonymus function with parameters
```
Comment (fwcd) [inline test/corpus/functions.txt:192]:
```suggestion
Anonymous function with parameters
```

File: test/corpus/functions.txt [hunk 3]
```diff
@@ -171,3 +171,106 @@ fun test() {
...
+    (variable_declaration
+      (simple_identifier))
+    (anonymous_function
+      (parameter
+        (simple_identifier)
+        (user_type
+          (type_identifier))))))
+
+==================
+Anonymus function with return type
```
Comment (fwcd) [inline test/corpus/functions.txt:216]:
```suggestion
Anonymous function with return type
```

File: test/corpus/functions.txt [hunk 4]
```diff
@@ -171,3 +171,106 @@ fun test() {
...
+      (type_identifier)))
+  (property_declaration
+    (variable_declaration
+      (simple_identifier))
+    (anonymous_function
+      (user_type
+        (type_identifier)))))
+
+==================
+Anonymus function with body
```
Comment (fwcd) [inline test/corpus/functions.txt:236]:
```suggestion
Anonymous function with body
```

**Comment by SalBakraa** [comment]
Marking the anonymous function as right associative has fixed the failing test.

**Comment by fwcd** [comment]
Thanks!

**Pull request merged**